### DISCUSSION TITLE
NVMe 固态硬盘与键值存储 KVell · SOSP 2019 - 面向信仰编程 · /papers-kvell

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#629] [Announcements]
https://draveness.me/papers-kvell/

**Comment by leafduo** [comment]
感觉生产上现在还用不上所谓『现代 SSD』，可能还得等个几年。

**Comment by draveness** [comment]
> 感觉生产上现在还用不上所谓『现代 SSD』，可能还得等个几年。

大规模落地确实要一段时间

**Comment by et-zhou** [comment]
捉个虫。
“很多数据库为了会使用特定的数据结构”这句里面的“为了”是不是应该改为“因此”？

---

*2020-10-19 UPDATES: 已修复*

**Comment by yuanjize** [comment]
数据库优化器是不是慢慢就没啥用了

**Comment by draveness** [comment]
> 数据库优化器是不是慢慢就没啥用了

那倒不会...

**Comment by maxnoodles** [comment]
新手程序员发问：这种线程分块的机制，扫描读的时候，是不是会需要查询携带某种特定的查询条件，不然会遍历所有线程。又因为是磁盘随机写，也会因此多读很多页(一页上可能就一条数据是需要的），导致内存消耗较大。

**Comment by draveness** [comment]
> 新手程序员发问：这种线程分块的机制，扫描读的时候，是不是会需要查询携带某种特定的查询条件，不然会遍历所有线程。

Redis、MySQL 这些存储不带查询条件也会遍历所有的数据啊，这个不是关键问题

> 又因为是磁盘随机写，也会因此多读很多页(一页上可能就一条数据是需要的），导致内存消耗较大。

内存里常驻的索引会帮助扫描数据，所以可以降低磁盘的随机读写次数

**Comment by jthmath** [comment]
我去年已经在生产环境中等规模用上nvme ssd了

**Comment by asddongmen** [comment]
提一个issue，推荐阅读里面的两个链接都是一样的，应该是贴错了。
1. https://draveness.me/papers-ceph
2. https://draveness.me/papers-ceph
